README.md

```diff
@@ -1,42 +1,589 @@
 # cloudify-nsx-plugin
 Cloudify Network Virtualization with VMware (NSX) plugin
 
-# Check platform example
-## Linux example
+# Common Documentation
+
+# Description
+
+The VMWare NSX plugin allows users to use a vCenter NSX based infrastructure for deploying services(both of network and secure) and applications.
+
+# Plugin Requirements
+
+* Python versions:
+    * 2.7.x
+* VMWare versions:
+    * NSX >= 6.2.3
+    * vSphere >= 6.0.0
+
+## NSX Environment
+
+* You will require a working vSphere and vCenter environment with installed NSX. The plugin was tested with version NSX 6.2.3.
+If you want to use network functionality - also required to use vSphere plugin 2.3.0+ version.
+
+# Types
+
+Each type has property `nsx_auth`. It can be used to pass parameters for authenticating.
+
+## nsx_auth fields:
+* `username`: user name for NSX
+* `password`: user password for nsx
+* `host`: nsx host
+* `raml`: optional, path to raml file, if not defined will be used embedded version from plugin.
+
+You can provide all properties described in node also as inputs for workflow action. E.g. if you don’t have nsx_auth as static properties values or can't provide it as inputs of blueprint,
+you also can use get_attributes call in place value in workflow input.
+
+```
+node_templates:
+  security_policy:
+    type: cloudify.nsx.security_policy
+    properties:
+      nsx_auth: <authentication credentials for nsx>
+         ....
+    interfaces:
+      cloudify.interfaces.lifecycle:
+        create:
+          inputs:
+            nsx_auth: <place for optional overwrite nsx_auth>
+              ...
+```
+
+General rules for properties - plugin always merge properties, inputs and overwrite if exist by values from runtime properties.
+
+As description for advanced properties please use [nsx pdf](https://pubs.vmware.com/NSX-62/topic/com.vmware.ICbase/PDF/nsx_62_api.pdf).
+
+## Security related functionality
+### cloudify.nsx.security_group
+
+A security group is a collection of assets or grouping objects from your vSphere inventory.
+
+**Derived From:** cloudify.nodes.ApplicationServer
+
+**Properties:**
+* `nsx_auth`: look above for information;
```

Comment:
Also, I'm not convinced that each of these should have semicolon line endings. Either full stop or nothing.